### DISCUSSION TITLE
Children components

### DISCUSSION THREAD

**Discussion by AEK-BKF** [rappasoft/laravel-livewire-tables#219] [Q&A]
Regarding the Bulk Actions, is there a way to put all components related to each action under the main component (table) as children ?
For example, if I have 3 actions where each action calls a modal component contains a form to do some tasks with selected records.
Working with children components is much easier than separated ones.
Thanks.

**Comment by rappasoft** [comment]
I'm not understanding fully what you are trying to accomplish. Maybe an actual example?

**Reply by AEK-BKF** [reply]
Okey, here a real example :
Let's take the package spatie-permissions, both User / Role can be assigned / revoked many permissions, 
So, in Permission tables, I need 4 Bulks actions like this :
```
public array $bulkActions = [
        'assignToUser' => 'Assign To User',
        'revokeFromUser' => 'Revoke From User',
        'assignToRole' => 'Assign To Role',
        'revokeFromRole' => 'Revoke From Role',       
    ];

    protected string $pageName = 'permissions';
    protected string $tableName = 'permissions';

    public function assignToUser()
    {
        if ($this->selectedRowsQuery->count() > 0)        
            $this->emit('adminAssignUserPermission', [$this->selectedRowsQuery->pluck('id')->toArray(), 'atu']);               
    }    

    public function revokeFromUser()
    {
        if ($this->selectedRowsQuery->count() > 0)        
            $this->emit('adminAssignUserPermission', [$this->selectedRowsQuery->pluck('id')->toArray(), 'rfu']);               
    }        

    public function assignToRole()
    {
        if ($this->selectedRowsQuery->count() > 0)        
            $this->emit('adminAssignRolePermission', [$this->selectedRowsQuery->pluck('id')->toArray(), 'atr']);               
    }    

    public function revokeFromRole()
    {
        if ($this->selectedRowsQuery->count() > 0)        
            $this->emit('adminAssignRolePermission', [$this->selectedRowsQuery->pluck('id')->toArray(), 'rfr']);               
    }
```
As you can see, I have two emits event which belongs to two components as Modal, each modal has a list of users or roles to be assigned/revoked, each component has a listener in order to show the modal having those selected rows in mount. 

and here the bade view : 
```
<livewire:admin.master.permissions.index />           
<livewire:admin.master.permissions.assignUser />            
<livewire:admin.master.permissions.assignRole /> 
```

The problem I'm facing with, only the first modal appears, i.e if I put (in blade) assignRole first, only this modal appears.
NB: no error appears in console or xhr.

May be you have an other approach to handle this.
Thanks.

**Reply by juanzu** [reply]
buddy, did you figure it out?
